### PR TITLE
LOK-2715  Save/update labels while creating monitoring policy. 

### PR DESCRIPTION
**Pull request by aurang-zeb313**
## Description
At the moment, when defining an alert rule in Cortex, labels are hard coded on the back end. As it saves the monitoring policy for the metrics threshold, the front end will now provide labels, expressions, and service types like SNMP and WSMAN. Labels must be added to the expression that is provided, as it will be transmitted from the front end.  Below is the modified request for saving metrics threshold in which labels, metricsExpression and serviceType is added.

{
		"policy": {
			"enabled": true,
			"memo": "test7",
			"name": "test7",
			"notifyByEmail": true,
			"notifyByPagerDuty": false,
			"notifyByWebhooks": false,
			"notifyInstruction": "",
			"rules": [
				{
					"alertConditions": [
						{
							"severity": "CRITICAL",
							"thresholdMetric": {
								"condition": ">",
								"enabled": true,
								"expression": "",
								"id": 1,
								"name": "criticalCondition",
								"threshold": 6
							},
							"triggerEvent": {
								"eventType": "METRIC_THRESHOLD",
								"id": 17270,
								"name": "HighThresholdExceeded",
								"uei": "uei.opennms.org/threshold/highThresholdExceeded"
							}
						},
						{
							"severity": "MAJOR",
							"thresholdMetric": {
								"condition": ">",
								"enabled": false,
								"expression": "",
								"id": 2,
								"name": "",
								"threshold": 0
							},
							"triggerEvent": {
								"eventType": "METRIC_THRESHOLD",
								"id": 17270,
								"name": "HighThresholdExceeded",
								"uei": "uei.opennms.org/threshold/highThresholdExceeded"
							}
						},
						{
							"severity": "MINOR",
							"thresholdMetric": {
								"condition": ">",
								"enabled": false,
								"expression": "",
								"id": 3,
								"name": "",
								"threshold": 0
							},
							"triggerEvent": {
								"eventType": "METRIC_THRESHOLD",
								"id": 17270,
								"name": "HighThresholdExceeded",
								"uei": "uei.opennms.org/threshold/highThresholdExceeded"
							}
						},
						{
							"severity": "WARNING",
							"thresholdMetric": {
								"condition": ">",
								"enabled": false,
								"expression": "",
								"id": 4,
								"name": "",
								"threshold": 0
							},
							"triggerEvent": {
								"eventType": "METRIC_THRESHOLD",
								"id": 17270,
								"name": "HighThresholdExceeded",
								"uei": "uei.opennms.org/threshold/highThresholdExceeded"
							}
						}
					],
					"componentType": "NODE",
					"detectionMethod": "EVENT",
					"eventType": "METRIC_THRESHOLD",
					"name": "testRule7",
          "serviceType":"SNMP",
					"thresholdMetricName": "cpuUtilization",
          "metricsExpression":" avg by (node_id) ((100 -(CpuRawIdle%1$s / (CpuRawIdle%1$s + CpuRawInterrupt%1$s + CpuRawUser%1$s +CpuRawWait%1$s +CpuRawNice%1$s + CpuRawSystem%1$s +CpuRawKernel%1$s + CpuRawSoftIRQ%1$s + CpuRawSteal%1$s + CpuRawGuest%1$s + CpuRawGuestNice%1$s)) * 100)) ",
          "labels":{"monitor":"SNMP"}
				}
			],
			"tags": [
				"default"
      ]
		}
	
}

## Jira link(s)
- https://opennms.atlassian.net/browse/LOK-2715

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

**Comment by aurang-zeb313** [comment]
I have validated and perform unit testing and its working as expected.

**Review by smunir-onms — COMMENTED** [review]
LGTM

File: metrics-threshold-processor/src/main/java/org/opennms/metrics/threshold/api/Constants.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * language governing permissions and limitations under the
+ * License.
+ */
+package org.opennms.metrics.threshold.api;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class Constants {
```
Comment (cgorantla) [inline metrics-threshold-processor/src/main/java/org/opennms/metrics/threshold/api/Constants.java:28]: do we still need these constants when we can get these from BFF. 

If it is first stage of adopting, then it is fine.

Comment (aurang-zeb313) [reply metrics-threshold-processor/src/main/java/org/opennms/metrics/threshold/api/Constants.java:28]: We  don't used it in the code. We will remove it but for now I want to keep it.

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**